### ISSUE TITLE
feat: disable RAG for model

### ISSUE THREAD

**Issue by juancarlosm** [open-webui/open-webui#7209]
# Feature Request

**Problem Description:**
When integrating Open-webui with agents or assistants (such as [Assistants Code Interpreter](https://platform.openai.com/docs/assistants/tools/code-interpreter)), we only need access to the uploaded file, not the content or RAG processing. I managed how to access the uploaded file via open-webui API, and then I can use the file in my agents, but the unnecessary RAG processing, especially for larger files, consumes time and resources. Additionally, each new query for the same chat triggers again RAG for embeddings and vector search, which is inefficient if not needed.

**Proposed Solution:**
Introduce a parameter in the model configuration to disable RAG processing for specific models. This would streamline operations by bypassing unnecessary file processing.

**Alternatives Considered:**
I attempted using functions with `self.file_handler = True`, but this approach did not prevent RAG processing.

**Related Issues:**
[Open-webui Issue #3556](https://github.com/open-webui/open-webui/issues/3556)

**Comment by oatmealm** [comment]
Similarly, and I'm not sure if it belongs here, when I've instructed a model to randomly pick facts from the RAG, I' don't need it to run a similarity query, I just need it to pass the documents and do the work it need to. I've been able to get some models to follow such a prompt (select a fact from the documents and ask me a question, then correct my reply and provide more information). I think right now I'll need to write a pipeline for this to work as expected). Often the content the rag pipelines returns is not complete and the model can't find all the information in what's been passed to it.

**Comment by CWrecker** [comment]
It would also be nice to turn this and vector storage off during file upload for unsupported files (e.g. MP4). 

 #6246

**Comment by vinismarques** [comment]
I am also very interested in having the option not to use RAG. Models can handle everything in the context window most of the time, especially the new Google models.

**Comment by EmilianoGarciaLopez** [comment]
Also very interested

**Comment by wangjiyang** [comment]
Need this feature also, I added a pipeline to process files, however this RAG feature is always trying to save vector information to vector db, which may takes very long time and failed for http request timeout. This feature is not always required since I need to process it by pipelines.

**Comment by tjbck** [comment]
You can set `BYPASS_EMBEDDING_AND_RETRIEVAL` to false from the admin settings.

**Comment by mkagit** [comment]
> You can set `BYPASS_EMBEDDING_AND_RETRIEVAL` to false from the admin settings.

It still appends the RAG task instruction to the system prompt even when BYPASS_EMBEDDING_AND_RETRIEVAL is false

**Comment by mkagit** [comment]
I want to send the attached file directly to the endpoint. Also when attaching an audio file it's triggers the transcribe engine. I want to send it to the endpoint.

**Comment by rgaricano** [comment]
![Image](https://github.com/user-attachments/assets/a0b1b3fe-5024-411c-8e7e-6079cb3d7646)

**Comment by ColbyB722** [comment]
> It still appends the RAG task instruction to the system prompt even when BYPASS_EMBEDDING_AND_RETRIEVAL is false

This is true. You can even see the RAG task instructions in the thinking tags of reasoning models like QwQ even when `BYPASS_EMBEDDING_AND_RETRIEVAL` is false

**Comment by marius10p** [comment]
I believe this behavior is still true, even when BYPASS_EMBEDDING_AND_RETRIEVAL is false. The attached document still gets processed by the RAG task, even if the output of the task is not used, which takes too long on every prompt in a chat context.

**Comment by dotmobo** [comment]
+1.

I added a model (GLM OCR) as a chat model and I want to be able to pass both images and PDFs to it. 

Images are working correctly. However, for PDFs, the GLM OCR model does not receive the actual file because the system uses Apache Tika for content extraction and a RAG process.

I want to keep the RAG and Apache Tika workflow for all other models, but disable it specifically for the GLM OCR model.

**Comment by Classic298** [comment]
@dotmobo would a file_handle filter do that job for you?
